src/server.ts

```diff
@@ -370,7 +370,7 @@ export class SubscriptionServer {
                   params.variables,
                   params.operationName);
 
-                if (!isAsyncIterable(promiseOrIterable) && promiseOrIterable instanceof Promise) {
+                if (!isAsyncIterable(promiseOrIterable) && isPromise(promiseOrIterable)) {
```

Comment:
I'd be more inclined to pull in and use the https://www.npmjs.com/package/is-promise library since it uses the more standard is thenable check.

Reply:
Hmm... yeah, that makes sense. Less code to maintain. I'll make the change in a few.